internal/datastore/crdb/schema/indexes.go

```diff
@@ -88,6 +88,19 @@ func IndexingHintForQueryShape(schema common.SchemaInformation, qs queryshape.Sh
 
 // IndexForFilter returns the index to use for a given relationships filter or nil if no index is forced.
 func IndexForFilter(schema common.SchemaInformation, filter datastore.RelationshipsFilter) *common.IndexDefinition {
+	// Special case: if the filter specifies the resource type and relation and the subject type and relation, then
+	// the schema diff index can be used.
+	if filter.OptionalResourceType != "" &&
+		filter.OptionalResourceRelation != "" &&
+		len(filter.OptionalSubjectsSelectors) == 1 &&
+		filter.OptionalSubjectsSelectors[0].OptionalSubjectType != "" &&
+		filter.OptionalSubjectsSelectors[0].RelationFilter.NonEllipsisRelation != "" &&
```

Comment:
Not a blocker, just an idea: I wish we could turn this into some fluent API helper